packages/frontend/src/pages/Playground.svelte

```diff
@@ -1,5 +1,151 @@
 <script lang="ts">
+import { conversations } from '../stores/conversations';
+import { studioClient } from '/@/utils/client';
+import {
+  isAssistantChat,
+  type ChatMessage,
+  type UserChat,
+  isPendingChat,
+  isUserChat,
+  type AssistantChat,
+} from '@shared/src/models/IPlaygroundMessage';
+import NavPage from '../lib/NavPage.svelte';
+import { playgrounds } from '../stores/playgrounds-v2';
+import { catalog } from '../stores/catalog';
+import Button from '../lib/button/Button.svelte';
+import { afterUpdate } from 'svelte';
+
 export let playgroundId: string;
+let prompt: string;
+let sendEnabled = false;
+let scrollable: Element;
+let lastIsUserMessage = false;
+let errorMsg = '';
+
+$: conversation = $conversations.find(conversation => conversation.id === playgroundId);
+$: playground = $playgrounds.find(playground => playground.id === playgroundId);
+$: model = $catalog.models.find(model => model.id === playground?.modelId);
+$: {
+  if (conversation?.messages.length) {
+    const latest = conversation.messages[conversation.messages.length - 1];
+    if (isAssistantChat(latest) && !isPendingChat(latest)) {
+      sendEnabled = true;
+    }
+    lastIsUserMessage = isUserChat(latest);
+  } else {
+    sendEnabled = true;
+  }
+}
+
+const roleNames = {
+  system: 'System',
+  user: 'User',
+  assistant: 'Assistant',
+};
+
+function getMessageParagraphs(message: ChatMessage): string[] {
+  if (isAssistantChat(message)) {
+    if (!isPendingChat(message)) {
+      return message.content?.split('\n') ?? [];
+    } else {
+      return message.choices
+        .map(choice => choice.content)
+        .join('')
+        .split('\n');
+    }
+  } else if (isUserChat(message)) {
+    const msg = message as UserChat;
+    return msg.content?.split('\n') ?? [];
+  }
+  return [];
+}
+
+function askPlayground() {
+  studioClient
+    .submitPlaygroundMessage(playgroundId, prompt)
+    .then(() => {
+      errorMsg = '';
+      sendEnabled = false;
+      prompt = '';
+    })
+    .catch((err: unknown) => {
+      errorMsg = String(err);
+      sendEnabled = true;
+    });
+}
+
+afterUpdate(() => {
+  if (!conversation?.messages.length) {
+    return;
+  }
+  const latest = conversation.messages[conversation.messages.length - 1];
+  if (isUserChat(latest) || (isAssistantChat(latest) && isPendingChat(latest))) {
+    scrollToBottom(scrollable);
```

Comment:
I feel weird about the scroll bottom, as we force the user to stay at the bottom every time we receive a token

https://github.com/projectatomic/ai-studio/assets/42176370/26449316-eab0-4d55-8c98-a4b60431756c

Reply:
@slemeur @jeffmaury wdyt? I tried to make the scroll as least disturbing as possible (by enabling it only when the assistant is replying), but it can be frustrating for the user if he tries to see the beginning of the conversation.

Reply:
This is not a priority for now, this can be fixed in a follow up PR, the error management is more important